client_test.go

```diff
@@ -2577,25 +2604,29 @@ func Test_Client_JobCompletion(t *testing.T) {
 		t.Parallel()
 
 		require := require.New(t)
-		var dbPool *pgxpool.Pool
 		now := time.Now().UTC()
-		config := newTestConfig(t, func(ctx context.Context, job *Job[callbackArgs]) error {
-			_, err := queries.JobSetState(ctx, dbPool, dbsqlc.JobSetStateParams{
+
+		client, bundle := setup(t, newTestConfig(t, nil))
+
+		type JobArgs struct {
+			JobArgsReflectKind[JobArgs]
+		}
+
+		AddWorker(client.config.Workers, WorkFunc(func(ctx context.Context, job *Job[JobArgs]) error {
+			_, err := client.driver.GetExecutor().JobUpdate(ctx, &riverdriver.JobUpdateParams{
 				ID:                  job.ID,
-				ErrorDoUpdate:       true,
-				Error:               []byte("{\"error\": \"oops\"}"),
+				ErrorsDoUpdate:      true,
+				Errors:              [][]byte{[]byte("{\"error\": \"oops\"}")},
```

Comment:
This one also discussed on FaceTime, but to clarify fo posterity: `JobUpdate` is only ever in tests, so performance doesn't matter much.

Also, since the introduction of `testfactory`, we're down to only seven usages of `JobUpdate` left. If we wanted to we could probably change things around and factor it out. Although, it does seem like one that we might need again in the future, so I didn't do that for now.